### PR TITLE
Parallelize put_chunk

### PR DESCRIPTION
**Pull request by arv**
There is no need to write data, meta and mutated_chunks serially

### CODE REVIEW

File: src/dag/write.rs [hunk 1]
```diff
@@ -43,22 +44,25 @@ impl<'a> Write<'_> {
...
-                .put(&Key::ChunkMeta(c.hash()).to_string(), meta)
-                .await?;
-        }
-
-        self.mutated_chunks
-            .write()
-            .await
-            .insert(c.hash().to_string());
+        let data_key = Key::ChunkData(c.hash()).to_string();
+        try_join!(
```
Comment (aboodman) [inline src/dag/write.rs:48]: I'm shocked you got this to work. Nice.

**Review by aboodman — COMMENTED** [review]
LGTM

File: src/dag/write.rs (hunk 1 above)
Comment (arv) [reply src/dag/write.rs:48]: The thing that made it easier was to extract the conditional one into a function at first... then I realized async blocks allows a concise syntax

**Comment by aboodman** [comment]
lgtm

**Pull request merged**